src/rpc/rpc_client.ts

```diff
@@ -784,6 +823,31 @@ export class RpcClient implements IClient {
   async getAuctionInfoByHash(
     blockHash: string
   ): Promise<StateGetAuctionInfoResult> {
+    try {
+      const resV2 = await this.getAuctionInfoV2ByHash(blockHash);
+      const result = new StateGetAuctionInfoResult();
+      result.auctionState = AuctionState.fromV2(resV2.auctionState);
+      result.version = resV2.version;
+      result.rawJSON = resV2.rawJSON;
+      return result;
+    } catch (err) {
+      const errorMessage = err?.message || '';
+      if (!errorMessage.includes('Method not found')) {
```

Comment:
@alexmyshchyshyn  getAuctionInfoV2ByHash can fail with next error for a block generated in Casper 1.x:

```
{"jsonrpc":"2.0","id":1,"error":{"code":-32003,"message":"Query failed","data":"value was not found in the global state"}}
```

the flow should continue in this case to use v1 method, right?